### PR TITLE
COMMERCE-9459 Shipments - Link to carrier based tracking

### PR DESCRIPTION
**Pull request by alessiorendina**
Base PR: https://github.com/liferay-commerce/liferay-portal/pull/2871

cc @luca-pellizzon 

https://issues.liferay.com/browse/COMMERCE-9459

### CODE REVIEW

**Comment by alessiorendina** [comment]
ci:test:sf

**Comment by alessiorendina** [comment]
ci:test:relevant

**Comment by alessiorendina** [comment]
ci:test:sf

**Comment by alessiorendina** [comment]
ci:test:relevant

**Comment by alessiorendina** [comment]
ci:test:sf

**Comment by alessiorendina** [comment]
ci:test:relevant

File: modules/apps/commerce/commerce-shipment-web/src/main/resources/META-INF/resources/commerce_shipment/edit_commerce_shipment_courier_detail.jsp [hunk 1]
```diff
@@ -18,6 +18,10 @@
 
 <%
 CommerceShipmentDisplayContext commerceShipmentDisplayContext = (CommerceShipmentDisplayContext)request.getAttribute(WebKeys.PORTLET_DISPLAY_CONTEXT);
+
+List<CommerceShippingMethod> commerceShippingMethods = commerceShipmentDisplayContext.getCommerceShippingMethods();
+
+CommerceShipment commerceShipment = commerceShipmentDisplayContext.getCommerceShipment();
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-shipment-web/src/main/resources/META-INF/resources/commerce_shipment/edit_commerce_shipment_courier_detail.jsp:24]: null check

File: modules/apps/commerce/commerce-shipment-web/src/main/resources/META-INF/resources/commerce_shipment/shipping_summary.jsp [hunk 2]
```diff
@@ -64,11 +59,11 @@ Format dateFormat = FastDateFormatFactoryUtil.getDate(DateFormat.MEDIUM, locale,
 					</span>
 
 					<c:choose>
-						<c:when test="<%= Validator.isBlank(carrier) %>">
+						<c:when test="<%= Validator.isBlank(commerceShipment.getCarrier()) %>">
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-shipment-web/src/main/resources/META-INF/resources/commerce_shipment/shipping_summary.jsp:62]: why don't we use the variable anymore?

File: modules/apps/commerce/commerce-shipment-web/src/main/resources/META-INF/resources/commerce_shipment/shipping_summary.jsp [hunk 3]
```diff
@@ -142,6 +181,11 @@ Format dateFormat = FastDateFormatFactoryUtil.getDate(DateFormat.MEDIUM, locale,
 				actionUrl=""
 				title='<%= LanguageUtil.get(request, "estimated-shipping-date") %>'
 			>
+
+				<%
+				Date shippingDate = commerceShipment.getShippingDate();
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-shipment-web/src/main/resources/META-INF/resources/commerce_shipment/shipping_summary.jsp:186]: null check

**Review by andrea-ale-sbarra — CHANGES_REQUESTED** [review]

**Comment by alessiorendina** [comment]
ci:test:sf

**Comment by alessiorendina** [comment]
ci.test:relevant

**Comment by alessiorendina** [comment]
ci:stop

**Comment by alessiorendina** [comment]
ci:test:sf

**Comment by alessiorendina** [comment]
ci:test:relevant

**Comment by alessiorendina** [comment]
@andrea-ale-sbarra I did all the requested changes, thanks!

**Comment by alessiorendina** [comment]
ci:test:relevant

**Comment by alessiorendina** [comment]
ci:test:relevant

**Comment by alessiorendina** [comment]
ci:test:relevant

**Comment by iovaneale** [comment]
forwarded to #2880

**Review by andrea-ale-sbarra — APPROVED** [review]

**Pull request closed without merging**